sdk/core/core-amqp/.eslintrc.json

```diff
@@ -1,7 +1,9 @@
 {
   "plugins": ["@azure/azure-sdk"],
-  "extends": ["../../.eslintrc.old.json", "plugin:@azure/azure-sdk/recommended"],
-  "parserOptions": {
-    "createDefaultProgram": true
+  "extends": ["plugin:@azure/azure-sdk/azure-sdk-base"],
+  "rules": {
+    "@typescript-eslint/no-unused-vars": "off",
+    "@azure/azure-sdk/ts-no-namespaces": "off",
```

Comment:
This rule states `TypeScript namespaces should not be used`, but in this PR modules are being changed to namespaces...

Reply:
Yeah, I don't think we should disable this rule. We need to convert the outdated `module`/`namespace` declarations to ES6 modules.

Reply:
We can log a separate issue to do the conversion, but then does the module need to change to namespace in this PR then?

Reply:
Those module to namespace conversions were done by eslint. What happened was: ESLint fixed what it could, then failed where it couldn't. I disabled it for those cases that failed the automation. I can revert those changes if that's better 🌞

Reply:
IIRC `namespace` should be equivalent to `module` (`module` has been totally scrubbed from all of the TypeScript documentation), so I think it would be okay to convert to namespace here and take an issue on modernizing the module structure to use ES6 modules.

Reply:
I am confused about your suggestion here. What do you mean by `convert to namespace` and `take an issue`?

Reply:
Discussed with @sadasant offline. We can leave as `namespace` for now and track restructuring the modules in a new issue, since it would likely be a pretty significant change.

Reply:
@willmtemple I've made a push that fixes the namespace problem and the no-inner-declarations problem! Your review is appreciated 🙌

Reply:
I've updated this PR after talking with Jeff! Please review the recent namespace changes 🌞